Add light and dark background settings to frontend configuration

mordilloSan opened the pull request:
**Description**

- Add lightBackground and darkBackground settings to frontend configuration. Closes #837 

**Additional Details**

- Introduces lightBackground and darkBackground to the Frontend struct.
- Introduces FallbackColor logic for robust color validation.
- Passes new background colors to the template and uses them in inline CSS.
- Adjusts HTML and CSS to use dynamic colors on first paint.
- Changed formatLine to account for double "#". This is for color definition.

**Additional Notes**

- Opted to only do a PR for main background because I know you are going to work on branding.
- Since the other parts of the layout follow specific CSS, if you are ok with the structure of this PR i can evolve to the other parts of the layout

gtsteffaniak commented:
Thanks for the PR and picking this up.

One of the challenges to a feature like this is creating custom HTML variables at runtime (not during compile). 

This implementation looks great with how it uses HTML template. Just a couple concerns:

1. What is the purpose for the generator and regex code? it looks like its just validating colors are valid HTML values? Does it have any other purpose besides validation?

2. I would recommend validating above once during startup in settings.go , then you can safely use that value in static.go `config.Frontend.DarkBackground`. The reason is that every page load would run the regex and validation logic unnecessarily. This process should happen only once during startup.


And personally, I would prefer not to default to a value silently, but instead fail or warn on startup with a log saying "invalid color provided". If you let the program run with invalid values the user might not understand why it didn't apply.

gtsteffaniak commented:
Another thought, there's been a few feature requests asking for custom CSS support, it seems a bit too small scoped to introduce this in frontend and not also offer CSS support.

What do you think about removing validation and offer a different feature: injecting CSS from a file provided into the index.html the same way.

No need to validate, if someone uses a custom CSS file, they are responsible for making sure its valid.

gtsteffaniak commented:
Yeah that makes sense to do both, for sure. regarding the follow up questions:

> For Docker and other deployments, does it make sense that we recommend mounting this custom CSS file specifically?

I think docker, I updated the recommended docker-compose to include a `data` directory with the config rather than mounting only the config file, this allows for adding whatever anyone wants to use, including a custom.css So no real changes needed. A custom CSS config option would just be a path to the file, like how database or other configs work.

> If we are going all-in on customization, I believe it make sense to move this config outside the frontend struct into its own css struct?

I don't think so. but Maybe all of this should go under a a new frontend key like `frontend.styling` to keep things organized. But in general, somewhere in frontend is the perfect place for this to go.

mordilloSan commented:
**PR updated.**

- Validation is now performed at startup as requested.

- Introduced logging to the validator.

- Updated comments for documentation generation.

- Default configuration now has proper defaults.

- Custom CSS file is loaded from either the project root `./custom.css` (added to .gitignore) or `/home/filebrowser/data/custom.css`.

- Restored `base.css` and `dark.css`. (Note: The config-generated styles use `!important` to ensure proper override.)

**Note:**
To ensure custom CSS rules are applied, the file should use `!important` where necessary.


Hope this hits the goal and let me know if further adjustment is needed
